### ISSUE TITLE
Proxy server support

### ISSUE THREAD

**Issue by LuciusV** [gorilla-devs/GDLauncher#1007]
**Is your feature request related to a problem? Please describe.**
I live in a country where using shadowsocks proxy is only a option to get reliable connection to curseforge and download mods.
I didn't found how to make GDLauncher use system HTTP or SOCKS5 proxy.

**Describe the solution you'd like**
Please add proxy server support for downloads and connections.

**Describe alternatives you've considered**
For example, MultiMC allows to use proxy server

**Comment by JonLit** [comment]
Our school wifi uses a proxyserver, too, and you can't get a connection to anything on the outside without using the proxy, I tried with proxychains, but it then stopped and I get this output:
`λ  proxychains gdlauncher
[proxychains] config file found: /etc/proxychains.conf
[proxychains] preloading /usr/lib/libproxychains4.so
fish: Job 1, 'proxychains gdlauncher' terminated by signal SIGTRAP (Verfolgungs- oder Haltepunkt erreicht)`

Update: Also doesn't work when running a console with proxychains and in this console trying to run gdlauncher, then I just get the line with `fish: Job 1, 'gdlauncher' terminated by signal SIGTRAP (Verfolgungs- oder Haltepunkt erreicht)`

More info about SIGTRAP: https://en.wikipedia.org/wiki/Signal_(IPC)#SIGTRAP

btw, the line is dependant on the console used, with bash I get the following: 
`Trace/Breakpoint ausgelöst (Speicherabzug geschrieben)`

I'm on Garuda Linux (Arch, Kernel 5.14.11-zen1-1-zen) GDLauncher v.1.1.14, proxychains-ng 4.15-1 (community Repo)

Update: Still not working, now with 5.15.13-zen1-1-zen

**Comment by Eskaan** [comment]
Wait so you have Linux on your schools PCs?
Finally a competent school caring about your data.

**Comment by JonLit** [comment]
> Wait so you have Linux on your schools PCs? Finally a competent school caring about your data.

no, sadly not, I use my private laptop, althogh they use linux on some servers and on the main entrance info screen I once saw the debian desktop

**Comment by huantianad** [comment]
@LuciusV @JonLit old issue but until this is implemented in the launcher, you can use a command line option to have GDL use a proxy. <https://www.electronjs.org/docs/latest/api/command-line-switches#--proxy-serveraddressport>
